src/basic_recipes/datashader.jl

```diff
@@ -0,0 +1,388 @@
+
+- `global_post::Function = Makie.equalize_histogram` function which gets called on the whole aggregation array before display (`global_post(final_aggregation_result)`).
+- `local_post::Function = identity` function which gets call on each element after aggregation (`map!(x-> local_post(x), final_aggregation_result)`).
+
+- `point_func::Function = identity` function which gets applied to every point before aggregating it.
+- `binfactor::Number = 1` factor defining how many bins one wants per screen pixel. Set to n > 1 if you want a corser image.
+- `show_timings::Bool = false` show how long it takes to aggregate each frame.
+- `interpolate::Bool = true` If the resulting image should be displayed interpolated.
+
+### Generic
+
+- `visible::Bool = true` sets whether the plot will be rendered or not.
+- `overdraw::Bool = false` sets whether the plot will draw over other plots. This specifically means ignoring depth checks in GL backends.
+- `transparency::Bool = false` adjusts how the plot deals with transparency. In GLMakie `transparency = true` results in using Order Independent Transparency.
+- `fxaa::Bool = true` adjusts whether the plot is rendered with fxaa (anti-aliasing).
+- `inspectable::Bool = true` sets whether this plot should be seen by `DataInspector`.
+- `depth_shift::Float32 = 0f0` adjusts the depth value of a plot after all other transformations, i.e. in clip space, where `0 <= depth <= 1`. This only applies to GLMakie and WGLMakie and can be used to adjust render order (like a tunable overdraw).
+- `model::Makie.Mat4f` sets a model matrix for the plot. This replaces adjustments made with `translate!`, `rotate!` and `scale!`.
+- `colormap::Union{Symbol, Vector{<:Colorant}} = :viridis` sets the colormap that is sampled for numeric `color`s.
+- `colorrange::Tuple{<:Real, <:Real}` sets the values representing the start and end points of `colormap`.
+- `nan_color::Union{Symbol, <:Colorant} = RGBAf(0,0,0,0)` sets a replacement color for `color = NaN`.
+- `lowclip::Union{Automatic, Symbol, <:Colorant} = automatic` sets a color for any value below the colorrange.
+- `highclip::Union{Automatic, Symbol, <:Colorant} = automatic` sets a color for any value above the colorrange.
+- `space::Symbol = :data` sets the transformation space for the plot
+"""
+@recipe(DataShader, points) do scene
+    Theme(
+
+        agg = AggCount(),
+        method = AggThreads(),
+        async_latest = true,
+        # Defaults to equalize_histogram
+        # just set to automatic, so that if one sets local_post, one doesn't do equalize_histogram on top of things.
+        global_post = automatic,
+        local_post = identity,
+
+        point_func = identity,
+        binfactor = 1,
+        show_timings = false,
+
+        colormap = theme(scene, :colormap),
+        colorrange = automatic
```

Comment:
IMO the colorrange calculation should be slightly changed here, so that we don't have to fiddle with colormaps.  Let the colorrange be (1, max) instead of (0, max), then set `lowclip = :transparent`, or `:black`, or what have you.  This should also make the colormap a lot more exact!